core/services/runs.go

```diff
@@ -120,7 +120,9 @@ func NewRun(
 	}
 
 	cost := &assets.Link{}
-	cost.Set(job.MinPayment)
+	if job.MinPayment != nil {
+		cost.Set(job.MinPayment)
+	}
```

Comment:
This works :+1: but looks to me like job.MinPayment should have a default value?

Reply:
It's assigned as 0 in the factory https://github.com/smartcontractkit/chainlink/blob/develop/core/store/models/job_spec.go#L98

This extra guard fixes an old job spec where `MinPayment` wasn't initialized.

Were you thinking that something else is required?